utils/ansible.py

```diff
@@ -0,0 +1,217 @@
+import fauxfactory
+import simplejson as json
+import os
+import subprocess
+import yaml
+from utils import conf
+from shutil import copyfile
+
+
+yml_path = os.path.dirname(__file__) + "/manageiq_ansible_module/"
+providers_basic_script = "providers_basic_script.yml"
+users_basic_script = "users_basic_script.yml"
+random_token = fauxfactory.gen_alphanumeric(906),
+random_miq_user = fauxfactory.gen_alphanumeric(8)
+
+
+def get_values_from_conf(provider, script_type):
+    # TODO Clean where possible
+    if script_type == 'providers':
+        return {
+            'name': 'Openshift 3',
+            'state': 'present',
+            'miq_url': conf.env.get("base_url", None),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+            'provider_api_hostname': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('hostname', []),
+            'provider_api_auth_token': conf.credentials['openshift-3'].token,
+            'hawkular_hostname': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('hostname', [])
+        }
+    elif script_type == 'users':
+        return {
+            'fullname': 'MIQUser',
+            'name': 'MIQU',
+            'password': 'smartvm',
+            'state': 'present',
+            'miq_url': conf.env.get("base_url", None),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+        }
+    elif script_type == 'custom_attributes':
+        return {
+            'entity_type': 'provider',
+            'entity_name': 'Openshift 3',
+            'miq_url': conf.env.get("base_url", None),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+        }
+
+
+# TODO Avoid reading files every time
+
+
+def read_yml(script, value):
+    with open(yml_path + script + ".yml", 'r') as f:
+            doc = yaml.load(f)
+    return doc[0]['tasks'][0]['manageiq_provider'][value]
+
+
+def get_yml_value(script, value):
+    with open(yml_path + script + ".yml", 'r') as f:
+            doc = yaml.load(f)
+    return doc[0]['tasks'][0]['manageiq_provider'][value]
```

Comment:
As you mentioned in the 'TODO' above, I think it'll be good to read the files to dict at the beginning, modify, and then write them back.
Also, read_yml and get_yml_value are duplicated.